codechain/keystore/HDStorageJson.py

```diff
@@ -47,3 +52,37 @@ def encode(seed: str, passphrase: str, meta: str):
         "seedHash": seed_hash,
         "meta": meta,
     }
+
+
+def decode(json: object, passphrase: str) -> str:
+    kdfparams = json["crypto"]["kdfparams"]
+    derivedKey = pbkdf2(
+        passphrase.encode("utf-8"),
+        bytes.fromhex(kdfparams["salt"]),
+        kdfparams["c"],
+        kdfparams["dklen"],
+        "sha256",
+    )
+    ciphertext = bytes.fromhex(json["crypto"]["ciphertext"])
+    mac = blake256(derivedKey[16:32] + ciphertext)
+
+    if mac != json["crypto"]["mac"]:
+        raise KeystoreError(ErrorCode.DECRYPTIONFAILED)
+
+    if json["crypto"]["cipher"] != "aes-128-ctr":
+        raise KeystoreError(ErrorCode.DECRYPTIONFAILED)
+
+    iv = bytes.fromhex(json["crypto"]["cipherparams"]["iv"])
+    ctr = Counter.new(128, initial_value=int_of_string(iv))
+    decipher = AES.new(derivedKey[0:16], AES.MODE_CTR, counter=ctr)
+    seed = decipher.decrypt(ciphertext)
+
+    seed_hash = blake256(seed)
+    if seed_hash != json["seedHash"]:
```

Comment:
Good! Can you check that whether you are comparing `bytes` and `str`?

Reply:
They are both `str`.